0.79 changelog (#31648)
microsoft/PowerToys: README.md

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.79 - February 2024 Update

In this release, we focused on stability and improvements.

**Highlights**

 - New feature: Keyboard Manager allows mapping shortcuts to start applications or opening URIs. Thanks [@jefflord](https://github.com/jefflord)!
 - New feature: Keyboard Manager allows shortcuts with chords. Thanks [@jefflord](https://github.com/jefflord)!
 - Modernized Color Picker with Fluent UX. Thanks [@niels9001](https://github.com/niels9001)!
 - Peek now is able to preview drives. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
 - File Locksmith has a Windows 11 tier 1 context menu.

### General

>  - Refactored code so that the English language is used as a fallback when a localized resource is not found.

Reviewer: ```suggestion
 - Refactored code so that the English is used as a fallback language when a localized resource can not be found.
```

Reviewer 2: Minus `the`

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

### Awake

>  - The setting reverts to "Keep using the current power plan" after Awake deactivates itself after any of the timed modes has expired.

Reviewer: ```suggestion
 - The setting now reverts to "Keep using the current power plan" after Awake deactivates itself after any of the timed modes has expired.
```

### Color Picker

 - Now uses WPFUI and the UI was updated to follow Fluent UX principles. Thanks [@niels9001](https://github.com/niels9001)!
 - Added enable and disable telemetry to align it with the other utilities.

### Command Not Found

 - Added telemetry for when a module instance is created in PowerShell.

### FancyZones

 - Fixed a memory leak occurring on work area changes.

### File Explorer add-ons

 - Added support to the .ksh, .zsh, .bsh and .env file types to Monaco previewer. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!

>  - Re-enabled the RendererAppContainer feature in WebView 2, since the associated crash has been fixed in the latest WebView 2 releases.

Reviewer: ```suggestion
 - Re-enabled the RendererAppContainer feature in WebView2, since the associated crash has been fixed in the latest WebView2 releases.
```

### File Locksmith

>  - Added a Windows 11 tier 1 context menu.

Reviewer: ```suggestion
 - Added as an entry in the Windows 11 tier 1 context menu.
```

### Hosts File Editor

 - Tweaked filter button style to indicate if filters are applied.
 - Added an error indicator to each input field to indicate why a new entry can't be created.

>  - Added a in-line delete button for each entry.

Reviewer: ```suggestion
 - Added an in-line delete button for each entry.
```

### Image Resizer

 - Units and resize modes are now localized.

>  - Fixed an issue causing the Height combo box to not hide when selecting percent as the measure unit in the Fit and Fill resize modes.

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

 - Tweaked and improved UI. Thanks [@Jay-o-Way](https://github.com/Jay-o-Way)!

### Keyboard Manager

 - Added a feature that allows remapping a shortcut to starting an application. Thanks [@jefflord](https://github.com/jefflord)!
 - Added a feature that allows remapping a shortcut to open a URI. Thanks [@jefflord](https://github.com/jefflord)!
 - Added chords to shortcuts. Thanks [@jefflord](https://github.com/jefflord)!
 - Send telemetry about the key/shortcut to key/shortcut remappings that are set. This doesn't include remap to text, application or URI since those might contain personal information.
 - Added telemetry to send a daily event that at least a key/shortcut to key/shortcut remapping was used.
 - Tweaked and fixed the chords code to better follow conventions when trying to call the same chord multiple times.

### Mouse Without Borders
 - Fixed an issue causing the target path string to be corrupted when registering as a service.

### Paste as Plain Text

>  - Prevent the windows menu from activating when the windows key is part of the activation shortcut and is released sooner.

Reviewer: ```suggestion
 - Prevent the start menu from activating when the windows key is part of the activation shortcut and is released sooner.
```

Reviewer 2: Caps

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

> ### Peek

Reviewer: Add "Added support to the .ksh, .zsh, .bsh and .env file types to Monaco previewer. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!" here too

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

 - Fixed a title bar issue after maximizing the Peek's window. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
 - Fixed a crash when trying to use Peek in File Explorer alternatives.
 - Added a previewer for drives. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

>  - File and folder previewers will now asynchronously calculate size, similar to the Properties screen in File Explorer. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Reviewer 3: This is actually only valid for folders

Author: Thank you! Pushed the change.

 - Optimized the code that gathers results from the plugin to reduce CPU consumption.
 - Optimized memory usage in the Window Walker plugin.
 - Fixed crashes and improved error handling when saving json configuration files.
 - The Program plugin will now correctly get the icon for a newly installed packaged application. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

### Quick Accent

 - Added support for the Slovenian character set. Thanks [@aklemen](https://github.com/aklemen)!

### Registry Preview

 - Fixed a crash when closing the application and the editor's right click menu is opened.

### Settings

 - Fixed an alignment issue in the flyout icons causing some icons to be centered when they shouldn't. Thanks [@niels9001](https://github.com/niels9001)!
 - Added the mention that Monaco supports .txt files. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!
 - Fixed an issue causing the Settings window to lose its previous maximized state. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

>  - Fixed a faulty XAML binding in the Text Extractor page.

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

### Documentation

 - Fixed broken links in doc/devdocs/readme.md. Thanks [@jem-experience](https://github.com/jem-experience)!

### Development

 - Updated Microsoft.MSBuildCache to 0.1.258-preview. Thanks [@dfederm](https://github.com/dfederm)!
 - Fixed CI to point VCToolsVersion to VC.CRT instead of VC.Redist version. Thanks [@snickler](https://github.com/snickler)!
 - Updated MSTest adapter and framework to 3.2.
 - Fixed CI by pointing WiX 3.14 urls and hashes to the latest release on GitHub.
 - Added Pro and Enterprise editions of Visual Studio to the repository's development configuration DSC scripts.
 - Updated CppWinRT to 2.0.240111.5.
 - Updated System.Drawing.Common to 8.0.2 to fix CI builds after the .NET 8.0.2 upgrade was released.

>  - Updated WPFUI version to 3.0.0. Thanks [@niels9001](https://github.com/niels9001)!

Reviewer 2: Indeed. Must have been mistaken with something else.

>  - XAML Styler is now fully enabled in the solution. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Reviewer 2: ... But what if a developer doesn't have it installed?

Author: Changed it to mention it's when CI runs.

Author: I've addressed the feedback in https://github.com/microsoft/PowerToys/pull/31648/commits/b3b0ee3f9c6233f9b79a9dbe0b9397cbab7a5136. Thank you!

 - Updated Microsoft.Web.WebView2 to 1.0.2365.46.

#### What is being planned for version 0.80

For [v0.80][github-next-release-work], we'll work on the items below:

 - Stability / bug fixes
 - Language selection
 - Automated UI testing through WinAppDriver

>  - Develop support for Desired State Configuration

Reviewer: ```suggestion
 - Develop support for Desired State Configuration
 - New module: File Actions Menu
```

;)
